app/ide-desktop/lib/dashboard/src/authentication/cognito.ts

```diff
@@ -367,19 +371,49 @@ export interface UserSession {
   readonly email: string
   /** User's access token, used to authenticate the user (e.g., when making API calls). */
   readonly accessToken: string
+  /** User's refresh token, used to refresh the access token when it expires. */
+  readonly refreshToken: string
+  /** URL to refresh the access token. */
+  readonly refreshUrl: string
+  /** Time when the access token will expire, date in ISO format(UTC). */
+  readonly expireAt: string
+  /** The client ID of the user pool. */
+  readonly clientId: config.UserPoolWebClientId
 }
 
 /** Parse a `CognitoUserSession` into a {@link UserSession}.
  * @throws If the `email` field of the payload is not a string. */
-function parseUserSession(session: cognito.CognitoUserSession): UserSession {
+function parseUserSession(
+  session: cognito.CognitoUserSession,
+  clientId: config.UserPoolWebClientId
+): UserSession {
   const payload: Readonly<Record<string, unknown>> = session.getIdToken().payload
   const email = payload.email
+
   /** The `email` field is mandatory, so we assert that it exists and is a string. */
   if (typeof email !== 'string') {
     throw new Error('Payload does not have an email field.')
   } else {
     const accessToken = session.getAccessToken().getJwtToken()
-    return { email, accessToken }
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const refreshUrl: string = session.getAccessToken().payload.iss
+    const expirationTimestamp = session.getAccessToken().getExpiration()
+
+    const expireAt = (() => {
```

Comment:
is the IIFE useful here?

Reply:
I used to initialize variables that way, it's like a shim for the `do expression`proposal: https://github.com/tc39/proposal-do-expressions

Reply:
makes sense... but given that it's not used anywhere else, i'd imagine we'd probably want to avoid it unless there's a compelling advantage

- the `(() =>` and `)()` are basically noise
- (minimal but nonzero) overhead of costructing the function, and of the function call